File: dmn-test-cases/signavio/dmn/dmn2java/expected/proto/proto3/java/dmn/example-credit-decision/AssessApplicantAgeRuleOutput.java

import java.util.*;

@javax.annotation.Generated(value = {"decisionTableRuleOutput.ftl", "assessApplicantAge"})
public class AssessApplicantAgeRuleOutput extends com.gs.dmn.runtime.RuleOutput {
    private java.math.BigDecimal assessApplicantAge;

    public AssessApplicantAgeRuleOutput(boolean matched) {
        super(matched);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("assessApplicantAge")
    public java.math.BigDecimal getAssessApplicantAge() {
        return this.assessApplicantAge;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("assessApplicantAge")
    public void setAssessApplicantAge(java.math.BigDecimal assessApplicantAge) {
        this.assessApplicantAge = assessApplicantAge;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        AssessApplicantAgeRuleOutput other = (AssessApplicantAgeRuleOutput) o;
        if (this.getAssessApplicantAge() != null ? !this.getAssessApplicantAge().equals(other.getAssessApplicantAge()) : other.getAssessApplicantAge() != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = 0;
        result = 31 * result + (this.getAssessApplicantAge() != null ? this.getAssessApplicantAge().hashCode() : 0);

        return result;
    }

    public String toString() {
        StringBuilder result_ = new StringBuilder("(matched=" + isMatched());
        result_.append(String.format(", assessApplicantAge='%s'", assessApplicantAge));
        result_.append(")");
        return result_.toString();
    }
}

File: dmn-signavio/src/test/resources/dmn2java/exported/proto/expected/proto3/kotlin/dmn/example-credit-decision/AssessApplicantAgeRuleOutput.kt

@javax.annotation.Generated(value = ["decisionTableRuleOutput.ftl", "assessApplicantAge"])
class AssessApplicantAgeRuleOutput(matched: Boolean) : com.gs.dmn.runtime.RuleOutput(matched) {
    @com.fasterxml.jackson.annotation.JsonProperty("assessApplicantAge")
    var assessApplicantAge: java.math.BigDecimal? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as AssessApplicantAgeRuleOutput
        if (if (this.assessApplicantAge != null) this.assessApplicantAge != other.assessApplicantAge else other.assessApplicantAge != null) return false

        return true
    }

    override fun hashCode(): Int {
        var result = 0
        result = 31 * result + (if (this.assessApplicantAge != null) this.assessApplicantAge.hashCode() else 0)

        return result
    }

    override fun toString(): String {
        val result_ = StringBuilder("(matched=" + isMatched())
        result_.append(String.format(", assessApplicantAge='%s'", assessApplicantAge))
        result_.append(")")
        return result_.toString()
    }
}
